src/repr/src/row.rs

```diff
@@ -713,6 +726,34 @@ where
 // public api
 
 impl Row {
+    /// Allocate an empty `Row` with a pre-allocated capacity.
+    #[inline]
+    pub fn with_capacity(cap: usize) -> Self {
+        Self {
+            data: SmallVec::with_capacity(cap),
+        }
+    }
+
+    /// Creates a new row from supplied bytes.
+    ///
+    /// # Safety
+    ///
+    /// This method relies on `data` being an appropriate row encoding, and can
+    /// result in unsafety if this is not the case.
+    pub unsafe fn from_bytes_unchecked(data: Vec<u8>) -> Self {
+        Row { data: data.into() }
+    }
+
+    /// Constructs a [`RowPacker`] that will pack datums into this row's
+    /// allocation.
+    ///
+    /// This method clears the existing contents of the row, but retains the
+    /// allocation.
+    pub fn packer(&mut self) -> RowPacker {
+        self.data.clear();
+        RowPacker { row: self }
+    }
```

Comment:
I'd be up for also having `fn packer_append` as well, unless we think it is an antipattern. There are use cases like "I am implementing `Map` and my job is to add just one column" where the stream of input `Row` objects would be great to re-use.

Reply:
> unless we think it is an antipattern

Not an antipattern, but definitely wrong to use in 99% of cases. A clunky name is probably the way to go, so that you don't reach for it accidentally but it is available if you need it.

Reply:
Added `RowPacker::for_existing_row` as a relatively inconvenient to type method that provides this functionality.